src/main/java/net/dv8tion/jda/bot/sharding/ShardManager.java

```diff
@@ -446,6 +451,85 @@ default Guild getGuildById(final String id)
+     * is made to the Discord servers.
+     *
+     * <p>The returned {@link net.dv8tion.jda.core.requests.RestAction RestAction} can encounter the following Discord errors:
+     * <ul>
+     *     <li>{@link net.dv8tion.jda.core.requests.ErrorResponse#UNKNOWN_USER ErrorResponse.UNKNOWN_USER}
+     *     <br>Occurs when the provided id does not refer to a {@link net.dv8tion.jda.core.entities.User User}
+     *     known by Discord. Typically occurs when developers provide an incomplete id (cut short).</li>
+     * </ul>
+     *
+     * @param  id
+     *         The id of the requested {@link net.dv8tion.jda.core.entities.User User}.
+     *
+     * @throws java.lang.NumberFormatException
+     *         If the provided {@code id} cannot be parsed by {@link Long#parseLong(String)}
+     * @throws java.lang.IllegalArgumentException
+     *         <ul>
+     *             <li>If the provided id String is null.</li>
+     *             <li>If the provided id String is empty.</li>
+     *         </ul>
+     *
+     * @return {@link net.dv8tion.jda.core.requests.RestAction RestAction} - Type: {@link net.dv8tion.jda.core.entities.User User}
+     *         <br>On request, gets the User with id matching provided id from Discord.
+     */
+    @CheckReturnValue
+    default RestAction<User> retrieveUserById(String id)
+    {
+        return retrieveUserById(MiscUtil.parseSnowflake(id));
+    }
+
+    /**
+     * Attempts to retrieve a {@link net.dv8tion.jda.core.entities.User User} object based on the provided id.
+     * <br>This first calls {@link #getUserById(long)}, and if the return is {@code null} then a request
+     * is made to the Discord servers.
+     *
+     * <p>The returned {@link net.dv8tion.jda.core.requests.RestAction RestAction} can encounter the following Discord errors:
+     * <ul>
+     *     <li>{@link net.dv8tion.jda.core.requests.ErrorResponse#UNKNOWN_USER ErrorResponse.UNKNOWN_USER}
+     *     <br>Occurs when the provided id does not refer to a {@link net.dv8tion.jda.core.entities.User User}
+     *     known by Discord. Typically occurs when developers provide an incomplete id (cut short).</li>
+     * </ul>
+     *
+     * @param  id
+     *         The id of the requested {@link net.dv8tion.jda.core.entities.User User}.
+     *
+     * @return {@link net.dv8tion.jda.core.requests.RestAction RestAction} - Type: {@link net.dv8tion.jda.core.entities.User User}
+     *         <br>On request, gets the User with id matching provided id from Discord.
+     */
+    @CheckReturnValue
+    default RestAction<User> retrieveUserById(long id)
+    {
+        // We need a JDA instance
+        JDA api = this.getShardCache().stream()
+                .findAny().orElseThrow(() -> new IllegalStateException("no active shards"));
+
+        // check cache
+        User user = this.getUserById(id);
+        if (user != null)
+            return new RestAction.EmptyRestAction<>(api, user);
```

Comment:
This should use the JDA instance of the user instead.

Reply:
Done